Python Page, Fonts

nendow02 opened the pull request:
Added fonts. Changed implementation of next button a little bit. Created python page with no functionality along with boolean dropdowns (which isn't 100% how I want it yet).

nendow02 committed:
Python Page, Fonts

Added fonts. Changed implementation of next button a little bit. Created python page with no functionality along with boolean dropdowns (which isn't 100% how I want it yet).

reginawang3495 commented on src/components/booleanDropdown.tsx:
```diff
@@ -0,0 +1,14 @@
+
+const options = ['True', 'False'];
+
+interface Props {
```
generally for ts react, the standard for these is `<Component>Props`!

reginawang3495 commented on src/components/booleanDropdown.tsx:
```diff
@@ -0,0 +1,14 @@
+}
+
+function BooleanDropdown(props:Props):JSX.Element  {
+  return <Dropdown className='drop' options={options} value='False' arrowClassName='arrow' onChange={() => props.change()}/>;
```
you can just do this!
```suggestion
  return <Dropdown className='drop' options={options} value='False' arrowClassName='arrow' onChange={props.change}/>;
```

reginawang3495 commented on src/pages/python.tsx:
```diff
@@ -0,0 +1,40 @@
+      <div id='problem'>
+        <div id='code'>
+          <div>
+            <span id='w'>num_of_antennas </span><span id='o'>= </span><span id='b'>2</span><br/>
+            <span id='w'>has_arms </span><span id='o'>= </span><span id='b'>True</span><br/>
+            <span id='w'>has_ears </span><span id='o'>= </span><span id='b'>False</span><br/>
+            <span id='w'>color </span><span id='o'>= </span><span id='b'>&quot;white&quot;</span><br/>
```
tbh this is a lil hard to read--can we do something like this?
```
            {Object.entries(outputAndMatches).forEach(([output, match]) => (
              <>
                <span id="w">{output}</span>
                <span id="o">= </span>
                <span id="b">{match}</span>
                <br />
              </>
            ))}
```

reginawang3495 replied:
and we can have this above!
```
const outputAndMatches = {
  num_of_antennas: "2",
  has_arms: "True",
  has_ears: "False",
  color: "white",
};
```

reginawang3495 commented on src/pages/python.tsx:
```diff
@@ -0,0 +1,40 @@
+import '../styles/python.scss';
```
since the `pages` are `components`, maybe the folder belongs in the `components` folder? Or if you don't want to do that, maybe it'd be good to rearrange so that the files make more sense for devs jumping in?

reginawang3495 replied:
for the current components, you could move them to a folder called `shared` or something!

reginawang3495 commented on src/styles/python.scss:
```diff
@@ -0,0 +1,54 @@
+  width: 450px;
+}
+
+#w {
```
these should probably be classes, not ids! (so `.w`)

see [id vs className](https://www.geeksforgeeks.org/difference-between-an-id-and-class-in-html/#:~:text=Difference%20between%20id%20and%20class,can%20apply%20to%20multiple%20elements.) for more info!

reginawang3495 replied:
can we also have it more descriptive like `white`? because i just saw w o b in the other place and was confused what it was

reginawang3495 commented on src/pages/python.tsx:
```diff
@@ -0,0 +1,40 @@
+import '../styles/python.scss';
+import BooleanDropdown from '../components/booleanDropdown';
+
+let i = '';
```
i dont really know what you were thinking, but should `i` be a react state?

we can also do it later lol since this is just a test

reginawang3495 reviewed (changes requested):
looks pretty good to me!! nice name hehe + i already like the color schema 

just some small dev nits :))

reginawang99 commented on .stylelintrc.json:
```diff
@@ -4,6 +4,7 @@
     "selector-no-qualifying-type": null,
-    "property-no-vendor-prefix": null
+    "property-no-vendor-prefix": null,
+    "selector-class-pattern": null
```
also, i'm totally down to disable some linting commands, but I think I just ran linting, and I think it's just for 
```
 Selector should be written in lowercase with hyphens
```
which seems like an ok standard? Thoughts?

nendow02 committed:
Review Changes

Made changes requested in code review.

reginawang3495 commented on src/components/pages/python.tsx:
```diff
@@ -0,0 +1,45 @@
+              </div>
+            ))}
+            <br/>
+            <span className='orange'>print</span><span className='white'>(num_of_antennas </span><span className='orange'>&gt; </span><span className='blue'>3 </span><span className='orange'>or </span><span className='white'>has_arms)</span><br/>
+            <span className='orange'>print</span><span className='white'>(</span><span className='orange'>not </span><span className='white'>has_ears)</span><br/>
+            <span className='orange'>print</span><span className='white'>(color </span><span className='orange'>!= </span><span className='blue'>&quot;red&quot;</span><span className='white'>)</span><br/>
```
can we possibly do the same thing as L6-L11? this is a little hard to read also!

reginawang3495 commented on src/components/pages/landing.tsx:
```diff
@@ -1,4 +1,4 @@
-import '../styles/landing.scss';
+import '../../styles/sidebar.scss';
```
just wondering, is this on purpose? :0 (`sidebar.scss` rather than `landing.scss`)

reginawang3495 reviewed (changes requested):
last things then im good to go!

nendow02 committed:
Update landing.tsx

Fixed one change

reginawang3495 commented on src/components/pages/python.tsx:
```diff
@@ -0,0 +1,45 @@
+function Python(): JSX.Element {
+
+  const outputAndMatches = {
+    num_of_antennas: '2',
+    has_arms: 'True',
+    has_ears: 'False',
+    color: '"white"',
+  };
+
+  return (
+    <div id='python'>
+      <h1>Match the output to the code:</h1>
+      <div id='problem'>
+        <div id='code'>
+          <div>
+            {Object.entries(outputAndMatches).map(([output, match]) => (
+              <div key={output}>
+                <span className='white'>{output}</span>
+                <span className='orange'> = </span>
+                <span className='blue'>{match}</span>
+              </div>
+            ))}
+            <br/>
+            <span className='orange'>print</span><span className='white'>(num_of_antennas </span><span className='orange'>&gt; </span><span className='blue'>3 </span><span className='orange'>or </span><span className='white'>has_arms)</span><br/>
+            <span className='orange'>print</span><span className='white'>(</span><span className='orange'>not </span><span className='white'>has_ears)</span><br/>
+            <span className='orange'>print</span><span className='white'>(color </span><span className='orange'>!= </span><span className='blue'>&quot;red&quot;</span><span className='white'>)</span><br/>
+          </div>
+        </div>
```
since we want this to be reusable, we could change to something like
create a `typings.ts` file with this
```
  interface bitOfCode {
    [key: string]: Color;
  }
  type lineOfCode = bitOfCode[];
```
and maybe a constants file with this (or jsut add it into this file)
```
  enum Color {
    White = "white",
    Orange = "orange",
    Blue = "blue",
  }
```
something like this for input 
```
const code: lineOfCode[] = [
    [
      { ["num_of_antennas"]: Color.White },
      { [" = "]: Color.Orange },
      { ["2"]: Color.Blue },
    ],
    [
      { ["has_arms"]: Color.White },
      { [" = "]: Color.Orange },
      { ["True"]: Color.Blue },
    ],
    [
      { ["has_ears"]: Color.White },
      { [" = "]: Color.Orange },
      { ["False"]: Color.Blue },
    ],
    [
      { ["color"]: Color.White },
      { [" = "]: Color.Orange },
      { ['"white"']: Color.Blue },
    ],
    [],
    [
      { ["print"]: Color.Orange },
      { ["(num_of_antennas"]: Color.White },
      { [" > "]: Color.Orange },
      { ["3"]: Color.Blue },
      { [" or "]: Color.Orange },
      { ["has_arms)"]: Color.White },
    ],
    ...
  ];
```
and something like this for the `JSX` part
```
<div id="code">
          {code.map((lineOfCode) => (
            <>
              {lineOfCode.map((bitOfCode) =>
                Object.entries(bitOfCode).map(([text, color]) => {
                  return <span className={color}>{text}</span>;
                })
              )}
              <br />
            </>
          ))}
        </div>
```

Then you can make the this part it's own component, which gets passed in linesOfCode so that this can be reusable in the future!

reginawang3495 replied:
Thus, this should likely be split into the `shared` folder or something

reginawang3495 reviewed (changes requested):
let me know if this makes sense!

nendow02 committed:
New code text implementation

Added reusable way to write multicolor code.

reginawang3495 reviewed (approved):
looks good to me!! nice reusable code for the future :))

reginawang3495 commented on src/components/shared/booleanDropdown.tsx:
```diff
@@ -0,0 +1,14 @@
+}
+
+function BooleanDropdown(props:BooleanDropdownProps):JSX.Element  {
+  return <Dropdown className='drop' options={options} value='False' arrowClassName='arrow' onChange={props.change}/>;
```
oops bug 
![image](https://user-images.githubusercontent.com/29292203/123505260-e71b1800-d612-11eb-91be-aea5aa02e2a9.png)

reginawang3495 reviewed (changes requested):
oops jk i found a bug--idk if you wanna fix it?

nendow02 committed:
Dropdown Fix

Dropdown now overlaps instead of pushing elements down. Also slightly changed implementation of typings.

reginawang3495 commented on src/components/shared/typings.tsx:
```diff
@@ -0,0 +1,31 @@
+      ))}
+    </div>
+  );
+}
```
nit: this could go into a `CodeFormat.tsx` file! Otherwise it'll be a bit hard to find

reginawang3495 commented on src/components/shared/typings.tsx:
```diff
@@ -0,0 +1,31 @@
+export type lineOfCode = bitOfCode[];
+
+interface CodeFormatProps {
+  text: lineOfCode[];
```
nit: i think `text` is a lil misleading since `text` generally denotes `string`, so i think maybe `code` would probably be better?

reginawang3495 commented on src/components/shared/typings.tsx:
```diff
@@ -0,0 +1,31 @@
+  White = 'white',
+  Orange = 'orange',
+  Blue = 'blue',
+}
```
you can probably put this in `constants.ts` instead since we won't have any types (if we remove the put the other stuff in `CodeFormat.tsx`) 

<small aside: if it doesn't have JSX, it doesn't need to `.tsx` ending--just `.ts` is enough>

reginawang3495 commented on src/components/shared/booleanDropdown.tsx:
```diff
@@ -0,0 +1,19 @@
+}
+
+interface BooleanDropdownProps {
+  change: () => void;
```
`on<Whatever>` is pretty standard for this!! It's to make it obvious in the naming that it's a method that is run when the drop down changes.
```suggestion
  onChange: () => void;
```

reginawang3495 commented on src/styles/booleanDropdown.scss:
```diff
@@ -0,0 +1,52 @@
+@import 'global';
```
nit: design feedback: i think the typing cursor is a little surprising tbh, something like
```
  cursor: pointer;
```
might be what you want!

reginawang3495 commented on src/styles/booleanDropdown.scss:
```diff
@@ -0,0 +1,52 @@
+}
+
+.arrow {
+  background: url('../assets/arrow.png') no-repeat right;
```
try to prefer svgs! they are more scalable + less data. In this case, it's not a huge deal, but still good to keep it consistent 

Here's how you can do it:
![image](https://user-images.githubusercontent.com/29292203/123525607-df955680-d686-11eb-9339-79a87d0dd5f4.png)

notice there isn't an overload of data:
```
<svg width="10" height="8" viewBox="0 0 10 8" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5 7.64285L0.669872 0.678567L9.33013 0.678567L5 7.64285Z" fill="#FF8A00"/>
</svg>
```

reginawang3495 reviewed (changes requested):
more nits, i think its good after this :)

nendow02 committed:
Small changes

Changes requested in most recent review

reginawang3495 reviewed (approved):
looks good to me!! 🥳

nendow02 committed:
Merge branch 'main' into Just-Python

nendow02 committed:
Merge Conflicts

Fixed conflicts with main.
